File: Difficulty: Easy/Geek Jump/geek-jump.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
    int minimumEnergy(vector<int>& h, int n) {
        // Code here
        int p1 = 0, p2=0;
        
        for(int i =1;i<n;i++){
            int l = p1 + abs(h[i]-h[i-1]);
            int r = INT_MAX;
            if(i>1) r = p2 + abs(h[i]-h[i-2]);
            
            int curr = min(l,r);
            p2 = p1;
            p1 = curr;
        }
        
        return p1;
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int N;
        cin >> N;
        vector<int> arr(N);
        for (int i = 0; i < N; i++) {
            cin >> arr[i];
        }
        Solution obj;
        cout << obj.minimumEnergy(arr, N) << "\n";
    }
    return 0;
}
// } Driver Code Ends

File: Difficulty: Easy/Geek Jump/geek-jump.py
#User function Template for python3

class Solution:
    def minimumEnergy(self, height, n):
        # Code here
        dp = [0] * n
        
        dp[0] = 0
        
        for i in range(1,n):
            left = dp[i-1] + abs(height[i]-height[i-1])
            
            right = float('inf')
            if(i>1):
                right = dp[i-2] + abs(height[i]-height[i-2])
            
            dp[i] = min(left,right)
            
        return dp[n-1]


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        n = int(input())
        height = list(map(int, input().split()))
        ob = Solution()
        print(ob.minimumEnergy(height, n))
        print("~")
# } Driver Code Ends